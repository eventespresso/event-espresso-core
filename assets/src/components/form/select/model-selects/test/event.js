import EventSelect from '../event';
import { shallow } from 'enzyme';

describe( 'EventSelect()', () => {
	it( 'matches snapshot with default props', () => {
		const wrapper = shallow( <EventSelect /> );
		expect( wrapper ).toMatchSnapshot();
	} );
	it( 'matches expected default props', () => {
		const wrapper = shallow( <EventSelect /> );
		expect( wrapper.prop( 'queryData' ) ).toEqual( {
			limit: 100,
			order: 'DESC',
			orderBy: 'start_date',
			showExpired: false,
		} );
		expect( wrapper.prop( 'selectConfiguration' ).placeholder ).toEqual(
			'Select Event...',
		);
		expect( wrapper.prop( 'modelName' ) ).toEqual( 'event' );
	} );
	it( 'does not allow overriding the modelName', () => {
		const wrapper = shallow( <EventSelect modelName={ 'tickets' } /> );
		expect( wrapper.prop( 'modelName' ) ).toEqual( 'event' );
	} );
} );
