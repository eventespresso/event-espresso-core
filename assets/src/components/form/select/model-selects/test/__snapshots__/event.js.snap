// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorEventSelect() matches snapshot with default props 1`] = `
<div
  class="components-base-control"
>
  <div
    class="components-base-control__field"
  >
    <label
      class="components-base-control__label"
      for="inspector-select-event-control-0"
    >
      Select Event
    </label>
    <span>
      EventSelect
    </span>
  </div>
</div>
`;

exports[`EventSelect() matches snapshot with default props 1`] = `
<Component
  getQueryString={[Function]}
  label="Select Event"
  modelName="event"
  optionsEntityMap={
    Object {
      "default": Object {
        "label": "EVT_name",
        "value": "EVT_ID",
      },
    }
  }
  queryData={
    Object {
      "limit": 100,
      "order": "DESC",
      "orderBy": "start_date",
      "showExpired": false,
    }
  }
  selectConfiguration={
    Object {
      "defaultValue": undefined,
      "loadingMessage": [Function],
      "noOptionsMessage": [Function],
      "onChange": undefined,
      "placeholder": "Select Event...",
    }
  }
/>
`;
